### PR TITLE
feat: [PL-26371]: Added new field Auth path in k8s auth for Vault

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
#### Screenshots

NA

Screenshot 2022-07-13 at 12 05 10 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by meenakshi-harness** [comment]
retrigger coverage

File: src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx [hunk 1]
```diff
@@ -143,6 +144,10 @@ const VaultConfigForm: React.FC<StepProps<StepDetailsProps> & ConnectorDetailsPr
             is: HashiCorpVaultAccessTypes.AWS_IAM,
             then: Yup.string().trim().required(getString('validation.regionRequired'))
           }),
+          k8sAuthEndpoint: Yup.string().when('accessType', {
+            is: HashiCorpVaultAccessTypes.K8s_AUTH,
+            then: Yup.string().trim()
+          }),
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx:150]: We can remove this as this field is optional now.

File: src/modules/35-connectors/interfaces/ConnectorInterface.ts [hunk 2]
```diff
@@ -117,6 +117,7 @@ export interface VaultConfigFormData {
   renewalIntervalMinutes: number
   vaultK8sAuthRole?: string
   serviceAccountTokenPath?: string
+  k8sAuthEndpoint: string
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/interfaces/ConnectorInterface.ts:120]: should be optional?

File: src/modules/35-connectors/pages/connectors/utils/__tests__/ConnectorUtils.test.tsx [hunk 3]
```diff
@@ -941,12 +942,125 @@ describe('Connector Utils', () => {
...
+            authToken: undefined,
+            appRoleId: undefined,
+            secretId: undefined,
+            useVaultAgent: false,
+            sinkPath: undefined,
+            secretEngineManuallyConfigured: false,
+            secretEngineName: 'harness',
+            secretEngineVersion: '2',
+            vaultK8sAuthRole: 'dummyRole',
+            k8sAuthEndpoint: 'dummy/k8s/login',
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/pages/connectors/utils/__tests__/ConnectorUtils.test.tsx:1057]: add a test case when k8sAuthEndpoint is not set.

**Review by meenakshi-harness — DISMISSED** [review]

**Review by reetika04 — APPROVED** [review]

**Pull request merged**